aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -306,8 +326,16 @@ private ApiCategory mockApiCategoryWithGraphQlApi() throws AmplifyException {
         when(mockApiPlugin.getPluginKey()).thenReturn(MOCK_API_PLUGIN_NAME);
         when(mockApiPlugin.getCategoryType()).thenReturn(CategoryType.API);
 
+        ApiEndpointStatusChangeEvent eventData =
+            new ApiEndpointStatusChangeEvent(ApiEndpointStatusChangeEvent.ApiEndpointStatus.REACHABLE,
```

Comment:
You could play a game here like:
```java
ApiEndpointStatuChangeEvent eventData =
    ApiEndpointStatus.REACHABLE.changeTo(ApiEndpiontStatus.UNKNOWN);
```

Basically, add afactory method to the enum, that has the ability to generate the change events. It knows the "from" based on what is being invoked, and it knows the "to", if we pass that as an argument.

Reply:
I think this one ended up getting addressed by some of the refactoring I did.